src/wp-includes/class-wp-user-query.php

```diff
@@ -840,7 +840,19 @@ public function query() {
 				$result->id = $result->ID;
 			}
 		} elseif ( 'all_with_meta' === $qv['fields'] || 'all' === $qv['fields'] ) {
-			cache_users( $this->results );
+			if ( function_exists( 'cache_users' ) ) {
+				cache_users( $this->results );
+			} else {
+				_doing_it_wrong(
+					'WP_User_Query',
```

Comment:
Arguably this should be `WP_User_Query::query`.

The following code will run just fine as there won't be an attempt to warm the cache until after the hook is run.

```php
$my_query = new WP_User_Query();
add_action( 'plugins_loaded', function() use ( $my_query ) {
  $my_query->prepare_query( /* args */ );
  $my_query->query();
} )
```

Reply:
Ah good point. Agreed that makes more sense.

Reply:
Is [`c0db2d1`](https://github.com/WordPress/wordpress-develop/pull/3557/commits/c0db2d1ad137bbc94849590a8fe4f66dad61b1ee) what you had in mind @peterwilsoncc?

Reply:
I think we should revert the change https://github.com/WordPress/wordpress-develop/pull/3557/commits/c0db2d1ad137bbc94849590a8fe4f66dad61b1ee. A warning will show even if you run a WP_User_Query fields => id.

Reply:
I might be misunderstanding @peterwilsoncc's suggestion. I thought that was the goal?

Reply:
I am not sure about the original feedback.

Reply:
👍 I'm fine reverting. Will wait for Peter's feedback.

Reply:
I was thinking of changing `WP_User_Query` to `WP_User_Query::query`.

I like the consistency of c0db2d1ad137bbc94849590a8fe4f66dad61b1ee warning for all user queries prior to `plugisn_loaded` but if it is reverted the messaging should be more precise: indicate the problem is querying full user objects rather than just IDs.

Reply:
> I was thinking of changing WP_User_Query to WP_User_Query::query.

Ah, my fault, 🤦 I understand now.

> I like the consistency of https://github.com/WordPress/wordpress-develop/commit/c0db2d1ad137bbc94849590a8fe4f66dad61b1ee warning for all user queries

Yeah in my mind while querying by ids wouldn't cause the fatal error, it still should be avoided as plugins haven't had a chance to properly setup filters yet either.